### DISCUSSION TITLE
Is it possible to update SageMaker Domain Sharing Settings?

### DISCUSSION THREAD

**Discussion by flexelem** [aws/aws-cdk#27018] [Q&A]
Hi there,

I created a sagemaker domain through aws-cdk. And now, I am trying to update its domain's sharing setting, however, I am getting an error;

```
Unable to update Domain [arn:aws:sagemaker:us-east-1:1111111111:domain/d-asdasd] with InService App. Delete all InService apps in the domain including shared Apps for [domain-shared] User Profile, and try again.
```

Moreover, the related CloudFormation [reference](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-sagemaker-domain-usersettings.html#cfn-sagemaker-domain-usersettings-sharingsettings) already points out that these attributes are specified at creation time of the resource but it is marked with `No Interruption` from CloudFormation reference. I want to be sure that these properties are not updatable?

Thanks!

**Comment by peterwoodworth** [comment]
Could you provide reproduction steps with minimal code sample please?

**Comment by flexelem** [comment]
Hi @peterwoodworth,

Here is the cdk sample for creating sagemaker domain. The attributes `vpcId`, `subnetIds`, and `executionRole` are defined before domain and I didn't shared the code to prevent verbosity.

```
new sagemaker.CfnDomain(this, 'SageMakerDomain', {
  authMode: 'SSO',
  domainName: `${cdk.Stage.of(this)!.stageName}-sagemaker-domain`,
  vpcId: props.vpcStack.vpc.vpcId,
  subnetIds: privateIsoSubnets.subnetIds,
  defaultUserSettings: {
    executionRole: sagemakerRole.roleArn,
    // sharingSettings is added after creating the domain
    sharingSettings: {
      notebookOutputOption: 'Allowed',
    },
  },
});
```

Additionally, here is the cli command and input json I used for updating `sharingSettings`

```
aws sagemaker update-domain --domain-id foobar --cli-input-json file://update-domain.json
```

```
{
    "DomainId": "foobar",
    "DefaultUserSettings": {
        "SharingSettings": {
            "NotebookOutputOption": "Allowed"
        }
    }
}
```
